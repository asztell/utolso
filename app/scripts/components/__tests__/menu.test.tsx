import React from "react";
import { IntlProvider } from "react-intl";
import { render } from "@testing-library/react";

import { Menu } from "../menu";

describe("<Menu />", () => {
  test("render Menu", () => {
    const { container } = render(
      // the IntlProvider would be extracted into a helper function
      // that would be used by all the tests
      <IntlProvider
        locale={navigator.language}
        messages={{
          // in a medium/large app the messages would be composed with a helper function
          // that would read the messages from a file
          // and merge them with the default text
          "Nav.NavItem.Sale": "test SALE UPON SALE",
          "Nav.NavItem.Holiday": "test HOLIDAY",
          "Nav.NavItem.New": "test NEW",
          "Nav.NavItem.Products": "test PRODUCTS",
          "Nav.NavItem.Best-Sellers": "test BEST-SELLERS",
          "Nav.NavItem.Goodbyes": "test GOODBYES",
          "Nav.NavItem.Inspiration": "test INSPIRATION",
          "PromoStrip.Promo.Text":
            "test COOL ADVERTISING WITH PROMOTIONS AND SALES!",
          "PromoStrip.URL.Text": "test SIGN UP",
          "SearchInput.Placeholder": "test SEARCH",
          "SearchResultsSummary.Displayed":
            "test DISPLAYING {displayed} OF {total} RESULTS",
          "SearchResultsSummary.Total": "test SEE ALL RESULTS",
        }}
      >
        <Menu />
      </IntlProvider>
    );
    expect(container).toMatchSnapshot();
  });
});
