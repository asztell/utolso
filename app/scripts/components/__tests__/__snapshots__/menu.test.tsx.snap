// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Menu /> render Menu 1`] = `
<div>
  <header
    class="Menu box"
  >
    <div
      class="MenuContent box"
    >
      <img
        alt="M.A.C. Cosmetics"
        class="Logo box"
        src="/img/mac_logo.svg
      "
      />
      <span
        class="PromoStrip box"
      >
        test COOL ADVERTISING WITH PROMOTIONS AND SALES!
        <a
          class="PromoURL"
          href="#"
        >
           
          test SIGN UP
        </a>
      </span>
      <div
        class="Search box"
      >
        <input
          placeholder="test SEARCH"
          type="text"
        />
      </div>
      <a
        class="MyMacLink"
        href="#"
      >
        <img
          alt="My MAC"
          src="/img/my_mac.png"
        />
      </a>
      <a
        class="ShoppingCartLink"
        href="#"
      >
        <img
          alt="Shopping Cart"
          src="/img/shopping_cart.png"
        />
      </a>
    </div>
    <nav
      class="Nav box"
    >
      <a
        class="NavItem NavItemRed"
        href="#"
      >
        test SALE UPON SALE
      </a>
      <a
        class="NavItem"
        href="#"
      >
        test HOLIDAY
      </a>
      <a
        class="NavItem"
        href="#"
      >
        test NEW
      </a>
      <a
        class="NavItem"
        href="#"
      >
        test PRODUCTS
      </a>
      <a
        class="NavItem"
        href="#"
      >
        test BEST-SELLERS
      </a>
      <a
        class="NavItem"
        href="#"
      >
        test GOODBYES
      </a>
      <a
        class="NavItem"
        href="#"
      >
        test INSPIRATION
      </a>
    </nav>
  </header>
</div>
`;
